### PR TITLE
Add option to block .eth usernames

### PR DESCRIPTION
**Pull request by Ghosts**
Adds an option to block users with .eth in their user names.

Not necessarily expecting this to get merged as I get that people may want to block NFT related things but not necessarily crypto as a whole - but I made these changes so I could personally start blocking `.eth` usernames & figured I'd throw it out there as an option for others!

### CODE REVIEW

File: src/content/autoBlock.ts [hunk 1]
```diff
@@ -118,6 +118,13 @@ interface User {
 	alreadyBlocked: boolean;
 }
 
+function blockEthUsername(userName?: string) {
+	console.log(userName);
+	return !settings?.actionOnEthUsernames || !userName
+		? false
+		: new RegExp('.eth', 'gi').test(userName);
```
Comment (ottomated) [inline src/content/autoBlock.ts]: This would block "beth" because of the `.`. You want `/\.eth$/`, probably

File: src/content/autoBlock.ts [hunk 2]
```diff
@@ -118,6 +118,13 @@ interface User {
 	alreadyBlocked: boolean;
 }
 
+function blockEthUsername(userName?: string) {
+	console.log(userName);
```
Comment (ottomated) [inline src/content/autoBlock.ts]: Remove console.log

Comment (ottomated) [reply src/content/autoBlock.ts]: Remove the settings check from here, it belongs in the callback with the rest

File: src/content/autoBlock.ts [hunk 3]
```diff
@@ -118,6 +118,13 @@ interface User {
 	alreadyBlocked: boolean;
 }
 
+function blockEthUsername(userName?: string) {
```
Comment (ottomated) [inline src/content/autoBlock.ts]: Rename to `isEthUsername`

**Review by ottomated — CHANGES_REQUESTED** [review]
Probably won't merge this regardless, but here are some bugs

File: src/content/autoBlock.ts (hunk 2 above)
Comment (Ghosts) [reply src/content/autoBlock.ts]: lol oops - should go unblock a few elizabeths, bethanys, and beths 😂

File: src/content/autoBlock.ts (hunk 1 above)
Comment (Ghosts) [reply src/content/autoBlock.ts]: lol oops - should go unblock a few elizabeths, bethanys, and beths 😂

**Pull request closed without merging**